### PR TITLE
[X86][GlobalISel] Reorganize br/brcond tests (NFC)

### PR DESCRIPTION
**Pull request by turinevgeny**
Removing duplicating tests under GlobalISel, consolidating to perform checks with all three selectors.

### CODE REVIEW

File: llvm/test/CodeGen/X86/isel-brcond-icmp.ll [hunk 1]
```diff
@@ -0,0 +1,1508 @@
+; NOTE: Assertions have been autogenerated by utils/update_llc_test_checks.py
+; RUN: llc < %s                               -mtriple=x86_64-apple-darwin10 -verify-machineinstrs | FileCheck %s --check-prefixes=SDAG-X64
+; RUN: llc < %s -fast-isel -fast-isel-abort=1 -mtriple=x86_64-apple-darwin10 -verify-machineinstrs | FileCheck %s --check-prefixes=FASTISEL-X64
```
Comment (RKSimon) [inline llvm/test/CodeGen/X86/isel-brcond-icmp.ll:3]: Some of these look like the X86/X64 variants can share a common prefix

Comment (e-kud) [reply llvm/test/CodeGen/X86/isel-brcond-icmp.ll:3]: Yes, the tests with a single argument are the same between X86/X64. I've recently thought that in many cases FastISel and SelectionDAG produce the same output, maybe we can try to unify them under something like `DAG-X86`, `DAG-X64` as they DAG compatible. Or another prefix to put them into the same basket if any.

Comment (RKSimon) [reply llvm/test/CodeGen/X86/isel-brcond-icmp.ll:3]: I have no objection if we want to use X86/X64 prefixes for DAG+FASTISEL - and just have GISEL as the special case, especially as we're aiming to remove the diffs entirely. 

Some of the GISEL diffs are just different LBB tags for some reason, which we can hopefully address pretty easily....

Comment (turinevgeny) [reply llvm/test/CodeGen/X86/isel-brcond-icmp.ll]: Makes sense, I made changes to the tests to share more common prefixes.

File: llvm/test/CodeGen/X86/isel-br.ll [hunk 2]
```diff
@@ -0,0 +1,31 @@
+; NOTE: Assertions have been autogenerated by utils/update_llc_test_checks.py UTC_ARGS: --version 4
+; RUN: llc < %s -O0 -mtriple=i686-linux-gnu                                     -verify-machineinstrs | FileCheck %s --check-prefix=DAG
```
Comment (arsenm) [inline llvm/test/CodeGen/X86/isel-br.ll:2]: don't really need all the -verify-machineinstrs

File: llvm/test/CodeGen/X86/isel-br.ll [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+; RUN: llc < %s -O0 -mtriple=i686-linux-gnu -fast-isel -fast-isel-abort=1       -verify-machineinstrs | FileCheck %s --check-prefix=DAG
+; RUN: llc < %s -O0 -mtriple=i686-linux-gnu -global-isel -global-isel-abort=1   -verify-machineinstrs | FileCheck %s --check-prefix=GISEL
+; RUN: llc < %s -O0 -mtriple=x86_64-linux-gnu                                   -verify-machineinstrs | FileCheck %s --check-prefix=DAG
```
Comment (arsenm) [inline llvm/test/CodeGen/X86/isel-br.ll:5]: use explicit -global-isel=0 for dag cases?

File: llvm/test/CodeGen/X86/isel-br.ll (hunk 2 above)
Comment (turinevgeny) [reply llvm/test/CodeGen/X86/isel-br.ll:2]: Do you mean for `isel-br.ll` file?

File: llvm/test/CodeGen/X86/isel-br.ll (hunk 3 above)
Comment (turinevgeny) [reply llvm/test/CodeGen/X86/isel-br.ll:5]: Added for SelectionDAG, or you meant FastISel as well?

Comment (arsenm) [reply llvm/test/CodeGen/X86/isel-br.ll]: I'm not sure what the flag interaction between -fast-isel and -global-isel is

**Review by arsenm — APPROVED** [review]

File: llvm/test/CodeGen/X86/isel-br.ll (hunk 3 above)
Comment (turinevgeny) [reply llvm/test/CodeGen/X86/isel-br.ll]: `TargetPassConfig.cpp` checks first checks for FastISel, then for GlobalISel.
https://github.com/llvm/llvm-project/blob/c95693c74601521463e12ff358fed6fb9ee736b9/llvm/lib/CodeGen/TargetPassConfig.cpp#L974
`-global-isel=0` probably makes sense for futureproofing purposes with SelectionDAG... don't have a preference here.

**Review by RKSimon — APPROVED** [review]
LGTM

**Pull request merged**